Fixes a bunch of hotkey preferences bugs

Gamer025 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This fixes the problem of the keybind conflict message being shown over and over again until you manually save your preferences with correct keybinds + resetting to classic keys works now
You had to manually save your prefs because the code didn't save the new unbound keys to the prefs file
Classic mode was broken because the emote hotkeys didn't have classic_keys set to Unbound by default

New hotkeys are now actually set to Unbound if no default key is set
This was broken for emote hotkeys because classic_keys = list("Unbound") was missing for them
It was also broken because the code assumed conflicting keys if some Unbound key already existed ...
The code also used classic key defaults even if you had hotkey mode enabled thats fixed now too

Fixes #59491 and fixes #56382
Closes #59490
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
I guess constantly getting a message to fix your keybinds is somewhat annoying but I honstly just don't want random "KEYBINDING CONFLICT IS STILL A THING" runtimes in the logs
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: The message for new duplicate keybinds should only appear once now
fix: Resetting your keybinds to classic mode works correctly now
fix: Hotkey mode is now actually used when new hotkeys with a default setting are added
fix: You will no longer get a duplicate keybinding conflict message for new hotkeys with no default key, they will just be silently set to unbound
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Gamer025 committed:
save new keybinds to preference file after setting them to unbound

Gamer025 committed:
change message to include keybinds without default key

Gamer025 committed:
fix resseting to classic hotkeys

tralezab commented on code/modules/client/preferences_savefile.dm:
```diff
@@ -127,11 +127,12 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 /datum/preferences/proc/announce_conflict(list/notadded)
 	to_chat(parent, "<span class='warningplain'><b><u>Keybinding Conflict</u></b></span>\n\
-					<span class='warningplain'><b>There are new <a href='?_src_=prefs;preference=tab;tab=3'>keybindings</a> that default to keys you've already bound. The new ones will be unbound.</b></span>")
+					<span class='warningplain'><b>There are new <a href='?_src_=prefs;preference=tab;tab=3'>keybindings</a> that default to keys you've already bound or that need to be manualy set. The new ones will be unbound.</b></span>")
```
```suggestion
					<span class='warningplain'><b>There are new <a href='?_src_=prefs;preference=tab;tab=3'>keybindings</a> that default to keys you've already bound or that need to be manually set. The new ones will be unbound.</b></span>")
```

LemonInTheDark commented:
He won

Gamer025 committed:
Update code/modules/client/preferences_savefile.dm

Co-authored-by: tralezab <40974010+tralezab@users.noreply.github.com>

cacogen commented:
Should announce_conflict() even be triggered by keybindings that need to be manually set? Doesn't that mean they're unbound by default?

Gamer025 commented:
> Should announce_conflict() even be triggered by keybindings that need to be manually set? Doesn't that mean they're unbound by default?

I think you are right and the code actually should be setting new keys to unbound (or their default value if provided) and only alert on conflict.
However since all the emote hotkeys were missing classic_keys = list("Unbound") the check_keybindings code was broken and didn't set the new emotes to Unbound.
This however begs the question what the "hotkeys" variable does as both codepaths here are identical in my opinion (the only difference here is the different name for the temporary for var):
![image](https://user-images.githubusercontent.com/33846895/120798321-7ade5480-c53d-11eb-9ee3-0581c7c47583.png)
Im quite sure the first codeblock should be `var/hotkeytobind in kb.hotkey_keys`
The fact that the hotkeys var was added by a commit without PR with only "keybindings save file fix" as description isn't very helpful.
Also im quite sure check_keybindings() also has another bug where it will not assign new hotkeys to Unbound if any other hotkey is already Unbound because it thinks that's a conflict ...
Why is this code broken in so many ways

Gamer025 committed:
fix check_keybindings
actually use hotkey defaults if in hotkey mode
save to pref files after checking keybinds
Change back alert message to only mentions conflicts

Gamer025 commented:
Ok all the issues should be fixed now, see the edited PR description / last commit message for what changed
